examples/3_Advanced_Topics/3-2_Creating_an_XML_BoM.py

```diff
@@ -0,0 +1,247 @@
+﻿# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:light
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.15.1
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# # Create an XML BoM using python
+
+# The `bom_types` namespace can be used in conjunction with the `BoMHandler` class to
+# create and manipulate BoMs for analysis. This example shows how this can be done, creating
+# a BoM for a laminated glass door and running a compliance query.
+
+# Most installations of Granta MI will use the default database key and table names:
+
+# +
+DB_KEY = "MI_Restricted_Substances"
+TABLE_NAME = "MaterialUniverse"
+# -
+
+# The structure of an XML BoM is hierarchical, individual parts belong to assemblies which can
+# belong to larger assemblies. It is possible to construct the BoM in one object, but it is often
+# simpler to build the part up from smaller assemblies. This example uses this approach.
+
+# It is possible to have the same part in multiple places in the BoM, so we define two helper functions
+# to create a copy of a part and set the quantity.
+
+# +
+import copy
+
+import ansys.grantami.bomanalytics.bom_types as bom_types
```

Comment:
We should probably say that this BoM schema is supported by Granta MI Restricted Substances and Sustainability reports 2025 R2 and later. Use the `eco2301` bom types if you want to work with an older version of the reports.

Either:
```suggestion
from ansys.grantami.bomanalytics.bom_types import eco2412
```

and make the corresponding changes throughout the script, or import eco2412 as bom_types. This might be confusing though, I often get questions where folks are confused about aliasing module names.